internal/pkg/skuba/addons/cilium_test.go

```diff
@@ -0,0 +1,286 @@
+		{
+			name:     "get cilium operator image without revision",
+			imageTag: "1.5.3",
+			want:     img.ImageRepository + "/cilium-operator:1.5.3",
+		},
+		{
+			name:     "get cilium operator image with revision",
+			imageTag: "1.5.3-rev2",
+			want:     img.ImageRepository + "/cilium-operator:1.5.3-rev2",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetCiliumOperatorImage(tt.imageTag); got != tt.want {
+				t.Errorf("GetCiliumOperatorImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestGetCiliumImage(t *testing.T) {
+	tests := []struct {
+		name     string
+		imageTag string
+		want     string
+	}{
+		{
+			name:     "get cilium image without revision",
+			imageTag: "1.5.3",
+			want:     img.ImageRepository + "/cilium:1.5.3",
+		},
+		{
+			name:     "get cilium image with revision",
+			imageTag: "1.5.3-rev2",
+			want:     img.ImageRepository + "/cilium:1.5.3-rev2",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			if got := GetCiliumImage(tt.imageTag); got != tt.want {
+				t.Errorf("GetCiliumImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderContext_CiliumInitImage(t *testing.T) {
+	type test struct {
+		name          string
+		renderContext renderContext
+		want          string
+	}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tt := test{
+			name: "render Cilium Init Image URL when cluster version is " + ver.String(),
+			renderContext: renderContext{
+				addon: Addon{
+					addon:             kubernetes.Cilium,
+					templater:         renderCiliumTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetCiliumImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: ver,
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: img.ImageRepository + "/cilium-init:([[:digit:]]{1,}.){2}[[:digit:]]{1,}(-rev[:digit:]{1,})?",
+		}
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.renderContext.CiliumInitImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			if !matched {
+				t.Errorf("renderContext.CiliumInitImage() = %v, want %v", got, tt.want)
+				return
+			}
+		})
+	}
+}
+
+func Test_renderContext_CiliumOperatorImage(t *testing.T) {
+	type test struct {
+		name          string
+		renderContext renderContext
+		want          string
+	}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tt := test{
+			name: "render Cilium Operator Image URL when cluster version is " + ver.String(),
+			renderContext: renderContext{
+				addon: Addon{
+					addon:             kubernetes.Cilium,
+					templater:         renderCiliumTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetCiliumImage},
+				},
```

Comment:
`CiliumOperatorImage()` does not need this, maybe we could drop this block.

Reply:
changed